### DISCUSSION TITLE
bug Continuous Trips

### DISCUSSION THREAD

**Discussion by SiNaPsEr0x** [jheredianet/Teslamate-CustomGrafanaDashboards#28] [Q&A] [answered]
this error on open page:

<img width="322" alt="image" src="https://user-images.githubusercontent.com/5438823/219323326-7b144976-d015-4ae4-8511-19f2a605f307.png">

**Comment by jheredianet** [comment, marked as answer]
Hi, division by zero means an incomplete drive in this case. 
It should be because you may have a wrong data recorded, meaning that does not have a start and end date. Check on the drives dashboard (from Teslamate) and look for any drive with not end date, or with zero or no duration, or no mileage recorder (a drive that has not been closed) as explained here: https://docs.teslamate.org/docs/maintenance/manually_fixing_data#terminate-a-drive-or-charge

**Reply by SiNaPsEr0x** [reply]
All drive have a start and end ... Can check with a command specific?

**Reply by jheredianet** [reply]
On thing you can do by browsing is to change the period as how on the image. For example, last week or last month until you find when you got that division by zero. You may also try with min distance value.

![image](https://user-images.githubusercontent.com/45735191/219618946-286dae98-8005-472d-a7b8-a234f1a0281e.png)

**Comment by SiNaPsEr0x** [comment]
I fixed the DB to delete drive without end date but same problem on select

**Reply by jheredianet** [reply]
Did you try what I've pointed on the previos message? Try to change the period to a short one to see what you get.

**Reply by SiNaPsEr0x** [reply]
With a shorter period it works

**Reply by jheredianet** [reply]
Perfect, so what you have to do is to open the period step by step (a week, a month or so) so you may find exactly which drive is causing you the division by zero.

**Reply by SiNaPsEr0x** [reply]
I order by null and nothing view

<img width="1034" alt="image" src="https://user-images.githubusercontent.com/5438823/220077070-734105cf-6394-4ab1-9cad-3e25e63d5631.png">

**Reply by jheredianet** [reply]
It's not enough, there are other tables implicated in the query. To make it easy,  try to do it using the "continuous trips dashboard"  as I explain you in my previos screenshot.

By the way, your query is wrong it should be:

SELECT * FROM drives WHERE end_date IS NULL;
